### DISCUSSION TITLE
😕Tedious Process of Developing a Custom Block

### DISCUSSION THREAD

**Discussion by boomyao** [toeverything/blocksuite#2355] [General]
First, I have to write the block trio(model, ui, service). 
Then navigate to different files to register them. 
Throughout the process, I also have to deal with non-intuitive type declarations.

**Comment by doodlewind** [comment]
Thanks for your feedback! The plan is to allow developing blocks with the framework you like (since you can embeds any framework into Web Component and just subscribe to model updates), and with declarative intuitive APIs. But at the moment we only have the Lit-based implementations, since we are focusing on improving editing experience at the moment.

P.S. one option is to build your editor (aka [host](https://blocksuite.affine.pro/editors-as-hosts.html)) UI *fully* with other framework by only reusing `@blocksuite/store`, this brings you maximum control. See https://github.com/zuozijian3720/blocksvite as an example.

**Reply by boomyao** [reply]
Through your introduction, I have noticed that BlockSuite seems to be a foundational framework for clients in the Web3.0 ecosystem🤩. 
I would like to clarify that my intention is to extend some custom components on top of the Affine software to implement additional functionalities, without duplicating the work that is currently being done by Affine.

**Reply by doodlewind** [reply]
Exactly, CRDT is decentralized by design, and we have plan to support P2P LAN synchronization in AFFiNE in the recent future. Feel free to share more thoughts about your ideas!
